--- a/net/minecraft/command/impl/FillCommand.java
+++ b/net/minecraft/command/impl/FillCommand.java
@@ -28,6 +28,8 @@
 import net.minecraft.util.text.TextComponentTranslation;
 import net.minecraft.world.WorldServer;
 
+import com.example.mod.FillLimitCommand;
+
 public class FillCommand
 {
     private static final Dynamic2CommandExceptionType TOO_BIG_EXCEPTION = new Dynamic2CommandExceptionType((p_208897_0_, p_208897_1_) ->
@@ -72,9 +74,9 @@
     {
         int i = area.getXSize() * area.getYSize() * area.getZSize();
 
-        if (i > 32768)
+        if (i > FillLimitCommand.fillLimit)
         {
-            throw TOO_BIG_EXCEPTION.create(32768, i);
+            throw TOO_BIG_EXCEPTION.create(FillLimitCommand.fillLimit, i);
         }
         else
         {
