--- a/net/minecraft/world/gen/layer/LayerUtil.java
+++ b/net/minecraft/world/gen/layer/LayerUtil.java
@@ -2,6 +2,9 @@
 
 import com.google.common.collect.ImmutableList;
 import java.util.function.LongFunction;
+
+import net.earthcomputer.bedrockified.BedrockAddOceanEdgeLayer;
+import net.earthcomputer.bedrockified.BedrockAddOceanTemperatureLayer;
 import net.minecraft.init.Biomes;
 import net.minecraft.util.registry.IRegistry;
 import net.minecraft.world.WorldType;
@@ -30,7 +33,8 @@
       IAreaFactory<T> iareafactory = p_202829_3_;
 
       for(int i = 0; i < count; ++i) {
-         iareafactory = parent.<T>apply((IContextExtended)contextFactory.apply(seed + (long)i), iareafactory);
+         // BEDROCK: adding a constant of 1 rather than i
+         iareafactory = parent.<T>apply((IContextExtended)contextFactory.apply(seed + 1), iareafactory);
       }
 
       return iareafactory;
@@ -45,7 +49,9 @@
       iareafactory = GenLayerAddIsland.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(50L), iareafactory);
       iareafactory = GenLayerAddIsland.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(70L), iareafactory);
       iareafactory = GenLayerRemoveTooMuchOcean.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(2L), iareafactory);
-      IAreaFactory<T> iareafactory1 = OceanLayer.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(2L));
+      // BEDROCK: rewrite ocean temperature generation
+      IAreaFactory<T> iareafactory1 = BedrockAddOceanTemperatureLayer.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(2L), iareafactory);
+      iareafactory1 = BedrockAddOceanEdgeLayer.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(2L), iareafactory1);
       iareafactory1 = repeat(2001L, GenLayerZoom.NORMAL, iareafactory1, 6, contextFactory);
       iareafactory = GenLayerAddSnow.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(2L), iareafactory);
       iareafactory = GenLayerAddIsland.INSTANCE.<T>apply((IContextExtended)contextFactory.apply(3L), iareafactory);
