--- a/net/minecraft/world/gen/feature/structure/ScatteredStructure.java
+++ b/net/minecraft/world/gen/feature/structure/ScatteredStructure.java
@@ -30,7 +30,7 @@
    protected boolean hasStartAt(IChunkGenerator<?> chunkGen, Random rand, int chunkPosX, int chunkPosZ) {
       ChunkPos chunkpos = this.getStartPositionForPosition(chunkGen, rand, chunkPosX, chunkPosZ, 0, 0);
       if (chunkPosX == chunkpos.x && chunkPosZ == chunkpos.z) {
-         Biome biome = chunkGen.getBiomeProvider().getBiome(new BlockPos(chunkPosX * 16 + 9, 0, chunkPosZ * 16 + 9), (Biome)null);
+         Biome biome = chunkGen.getBiomeProvider().getBiome(new BlockPos(chunkPosX * 16 + 8, 0, chunkPosZ * 16 + 8), (Biome)null); // BEDROCK: +8, not +9
          if (chunkGen.hasStructure(biome, this)) {
             return true;
          }
