--- a/net/minecraft/world/gen/feature/structure/StrongholdStructure.java
+++ b/net/minecraft/world/gen/feature/structure/StrongholdStructure.java
@@ -51,7 +51,7 @@
    }
 
    protected StructureStart makeStart(IWorld worldIn, IChunkGenerator<?> generator, SharedSeedRandom random, int x, int z) {
-      Biome biome = generator.getBiomeProvider().getBiome(new BlockPos((x << 4) + 9, 0, (z << 4) + 9), Biomes.DEFAULT);
+      Biome biome = generator.getBiomeProvider().getBiome(new BlockPos((x << 4) + 8, 0, (z << 4) + 8), Biomes.DEFAULT); // BEDROCK: +8, not +9
       int i = 0;
 
       StrongholdStructure.Start strongholdstructure$start;
