--- a/net/minecraft/util/SharedSeedRandom.java
+++ b/net/minecraft/util/SharedSeedRandom.java
@@ -1,15 +1,17 @@
 package net.minecraft.util;
 
+import net.earthcomputer.bedrockified.BedrockRandom;
+
 import java.util.Random;
 
-public class SharedSeedRandom extends Random {
+public class SharedSeedRandom extends BedrockRandom { // BEDROCK: changed RNG
    private int usageCount;
 
    public SharedSeedRandom() {
    }
 
    public SharedSeedRandom(long seed) {
-      super(seed);
+      super((int) seed); // BEDROCK: 32-bit seeds
    }
 
    public void skip(int bits) {
@@ -47,8 +49,9 @@
 
    public long setLargeFeatureSeed(long seed, int x, int z) {
       this.setSeed(seed);
-      long i = this.nextLong();
-      long j = this.nextLong();
+      // BEDROCK: replace nextLongs with nextInts
+      long i = this.nextInt();
+      long j = this.nextInt();
       long k = (long)x * i ^ (long)z * j ^ seed;
       this.setSeed(k);
       return k;
@@ -61,6 +64,6 @@
    }
 
    public static Random seedSlimeChunk(int p_205190_0_, int p_205190_1_, long p_205190_2_, long p_205190_4_) {
-      return new Random(p_205190_2_ + (long)(p_205190_0_ * p_205190_0_ * 4987142) + (long)(p_205190_0_ * 5947611) + (long)(p_205190_1_ * p_205190_1_) * 4392871L + (long)(p_205190_1_ * 389711) ^ p_205190_4_);
+      return new BedrockRandom((p_205190_0_ * 522133279) ^ p_205190_1_); // BEDROCK
    }
 }
